exasol_script_languages_container_ci_setup/lib/run_generate_buildspec.py

```diff
@@ -0,0 +1,94 @@
+
+    def __str__(self):
+        return f"Flavor(original={self.flavor_original}, formatted={self.flavor_formatted})"
+
+    def __repr__(self):
+        return self.__str__()
+
+
+def get_pip_location_for_pkg(dependent_pkg: str):
+    self_pkg_requirements = pkg_resources.working_set.by_key["script-languages-container-ci-setup"].requires()
+    searched_pgk_url = [pkg.url for pkg in self_pkg_requirements if pkg.name == dependent_pkg]
+    if len(searched_pgk_url) == 0:
+        raise RuntimeError(f"Missing dependency to package '{dependent_pkg}'")
+    elif len(searched_pgk_url) > 1:
+        raise RuntimeError(f"Multiple dependency entries found for package '{dependent_pkg}'")
+    return searched_pgk_url[0]
+
+
+def validate_config_file(config_file: Optional[str]):
+    if config_file is not None:
+        with open(config_file, "r") as config_file_:
+            config = json.load(config_file_)
+            config_schema = json.loads(render_template("config_schema.json"))
+            jsonschema.validate(instance=config, schema=config_schema)
+            ignored_folders = config["build_ignore"]["ignored_folders"]
+            for folder in ignored_folders:
+                folder_path = Path(folder)
+                if not folder_path.exists():
+                    raise ValueError(f"Ignored folder '{folder}' does not exist.")
+
+
+def get_config_file_parameter(config_file: Optional[str]):
+    if config_file is not None:
+        return f"--config-file {config_file}"
+    else:
+        return ""
+
+
+def run_generate_buildspec(
+        flavor_root_paths: Tuple[str, ...],
+        output_pathname: str,
+        config_file: Optional[str]):
+    validate_config_file(config_file)
+    flavors = set()
+    logging.info(f"Run run_generate_buildspec for paths: {flavor_root_paths}")
+    for flavor_root_path in [Path(f).resolve() for f in flavor_root_paths]:
+        assert flavor_root_path.is_dir()
+        assert flavor_root_path.exists()
+        assert flavor_root_path.name == "flavors"
+        for child in flavor_root_path.iterdir():
+            if child.is_dir():
+                flavors.add(Flavor(child.name))
+    logging.info(f"Found flavors: {flavors}")
+    buildspec_body = []
+    for flavor in flavors:
+        buildspec_body.append(render_template("buildspec_batch_entry.yaml",
+                                                 flavor_original=flavor.flavor_original,
+                                                 flavor_formatted=flavor.flavor_formatted,
+                                                 out_path=output_pathname))
+
+    result_yaml = render_template("buildspec_hull.yaml", batch_entries="\n".join(buildspec_body))
+
+    output_pathname = Path(output_pathname)
+    with open(output_pathname / "buildspec.yaml", "w") as output_file:
+        output_file.write(result_yaml)
+
+    script_languages_ci_location = get_pip_location_for_pkg("script-languages-container-ci")
```

Comment:
This needs explanation and do we really need pkg_resources for get_pip_location_for_pkg and for what do we need the package location

Reply:
As discussed offline, I will change the logic so that it uses the poetry environment of the SLC project to run the CI build. With that we get the implicit dependency to `script-languages-container-ci`, the one defined in the toml file of the SLC project!